src/Stryker.Core/Stryker.Core.UnitTest/Reporters/Progress/ProgressBarReporterTests.cs

```diff
@@ -54,7 +57,61 @@ public void ReportRunTest_ShouldReportTestProgressAs50PercentageDone_And_FirstTe
                                                               y => ((string)y.ElementAt(0)).EndsWith("-----") &&
                                                                    (int)y.ElementAt(1) == 1 &&
                                                                    (int)y.ElementAt(2) == 2 &&
-                                                                   (int)y.ElementAt(3) == 50)));
+                                                                   (int)y.ElementAt(3) == 50 &&
+                                                                   ((string)y.ElementAt(4)).Equals("~0m 00s"))));
+        }
+
+        [Fact]
+        public void ReportRunTest_ShouldReportTestProgressAs50PercentageDone_And_FirstTestExecutionTimeInMinutes()
+        {
+            _progressBarReporter.ReportInitialState(10000);
+            Thread.Sleep(10);
```

Comment:
Well...
I am not found of alternatives (I can think of):
1) testing time formatting in isolation: would not test time estimation at all
2) abstracting the time source: would require additional tests for the time source, tests that would logically require some time control (via Thread.Sleep)

If pressed to pick one, I would favour (2) as it offer an entry point to work on alternative time estimation strategy. But getting read of every *Thread.Sleep* looks difficult.
 I will try to make a proposal in that direction,,,

Reply:
What about an abstraction like:
https://stackoverflow.com/questions/2425721/unit-testing-datetime-now
It allows us to create a test for specific timestamps, like when 10 minutes passed, or even 10 hours passed. I think it's ok to exclude this from unit testing since it's only logic is passing the current time.

Reply:
Done